### PR TITLE
fix: Language chosen in user preferences persists (#4310)

### PR DESCRIPTION
**Pull request by bmihelac**

### CODE REVIEW

File: wagtail/admin/auth.py [hunk 1]
```diff
@@ -154,13 +155,19 @@ def decorated_view(request, *args, **kwargs):
...
+                if language != get_language():
+                    original_language = get_language()
+                    activate_lang(language)
                 time_zone = user.wagtail_userprofile.get_current_time_zone()
                 activate_tz(time_zone)
-            return view_func(request, *args, **kwargs)
+            response = view_func(request, *args, **kwargs)
+            if original_language:
+                activate_lang(original_language)
+            return response
```
Comment (gasman) [inline wagtail/admin/auth.py]: Could we use `django.utils.translation.override` here, as you've done for the `language_preferences` view? It seems a lot neater, and I have more confidence that Django's implementation will handle the edge cases correctly (e.g. what happens if the view throws an exception) than our own code.

Comment (bmihelac) [reply wagtail/admin/auth.py]: good point, I'll update ASAP

Comment (bmihelac) [reply wagtail/admin/auth.py]: updated

**Comment by gasman** [comment]
Thanks @bmihelac! Merged in df1608b08031d259970943adff87d2d078b77b68.

**Comment by bmihelac** [comment]
Thanks for review and merging @gasman

**Pull request closed without merging**